Dockerfile_mbtci_template

```diff
@@ -135,39 +147,74 @@ RUN ARCH= && dpkgArch="$(dpkg --print-architecture)" \
     && curl -fsSLO --compressed https://storage.googleapis.com/golang/go${GO_VERSION}.${GOOS}-${ARCH}.tar.gz \
     && curl -fsSLO --compressed https://storage.googleapis.com/golang/go${GO_VERSION}.${GOOS}-${ARCH}.tar.gz.asc \
     && gpg --batch --verify go${GO_VERSION}.${GOOS}-${ARCH}.tar.gz.asc go${GO_VERSION}.${GOOS}-${ARCH}.tar.gz \
-    && tar -xvf go${GO_VERSION}.${GOOS}-${ARCH}.tar.gz -C /usr/local \
+    && mkdir -p ${GO_HOME} \
+    && tar -xvf go${GO_VERSION}.${GOOS}-${ARCH}.tar.gz -C ${GO_HOME} --strip-components=1 \
+    && mkdir -p ${GOPATH} \
+    && ln -s ${GO_HOME} ${GOPATH} \
+    && chmod -R 777 ${GOPATH} \
```

Comment:
The original shell code was working perfectly without introducing any permission security issues, unneeded symlinks, unneeded and unused for building env variables. Please revert. 

cf. previous comment.

Reply:
In the past implementation before we merge all Docker files into Dockerfile_mbtci_template, $GOPATH path is set 777 permission, I am afraid some user may reply on the write permission, so I rever that code.

You can ref : https://github.com/SAP/cloud-mta-build-tool/commit/35393669a1319333777bf26c864976307a496f0b#diff-3632a131a98f4a6db44b6238249146e491fa94be0b683a407b11498c3bffc11c

And, for the new added GO_HOME env, you are right, I will change it to a local var and will not export it as a ENV

Reply:
I revert the code which change go install path to "/usr/local/go", now go is installed under "/usr/local"

https://github.com/SAP/cloud-mta-build-tool/pull/1023/commits/b3ea0b94387d71af5ac4e4952a133f0a78328619